packages/zoe/src/contracts/priceAggregator.js

```diff
@@ -231,19 +241,75 @@ const start = async zcf => {
       });
       ({ priceAuthority, adminFacet: priceAuthorityAdmin } = paKit);
     },
-    async initOracle(oracleInstance, query = undefined) {
+    makeOracleInvitation: async () => {
+      /** @type {OfferHandler<OracleAdmin>} */
+      const offerHandler = async (
+        seat,
+        { notifier: oracleNotifier, scaleValueOut = 1 } = {},
+      ) => {
+        const admin = await creatorFacet.initOracle();
+        seat.exit();
+        if (!oracleNotifier) {
+          // No notifier to track, just let them have the direct admin.
+          return admin;
+        }
+
+        // Adapt the notifier to push results.
+        const recurse = ({ value, updateCount }) => {
```

Comment:
I'm not clear on what `value` can be. A type annotation would help. Is `value` readonly?